### PR TITLE
feat: adds package level logger support, that defaults to console logger

### PR DESCRIPTION
**Pull request by nxtcoder17**
## Summary by Sourcery

Introduce a package-level default logger with console-based fallback and update usage to go through this logger.

New Features:
- Add package-level logging functions that delegate to a configurable default logger.
- Provide helpers to obtain derived loggers (With, Slog, Clone) from the package-level logger.

Enhancements:
- Replace the prior global default logger mechanism with a new package-level logger that defaults to the console logger.

Tests:
- Add tests verifying dynamic switching of the package-level default logger, its logging methods, and nil-safety behavior when resetting the logger.

### CODE REVIEW

File: package-logger.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	logger = l
+}
+
+var Debug = logger.Debug
+var Info = logger.Info
+var Warn = logger.Warn
+var With = logger.With
+var Slog = logger.Slog
+var Clone = logger.Clone
+var Error = logger.Error
```
[the remark this answers is not in the document]
Comment (nxtcoder17) [reply package-logger.go]: Thanks, i have updated it in the followup commit

**Comment by nxtcoder17** [comment]
@sourcery-ai review

**Comment by nxtcoder17** [comment]
@sourcery-ai dismiss

**Comment by nxtcoder17** [comment]
@sourcery-ai dismiss

**Comment by nxtcoder17** [comment]
@sourcery-ai review

**Comment by nxtcoder17** [comment]
@sourcery-ai review

**Comment by nxtcoder17** [comment]
@sourcery-ai dismiss

**Comment by nxtcoder17** [comment]
@sourcery-ai review
